### PR TITLE
Add the py-amqp package

### PR DESCRIPTION
**Pull request by koning**
This package is a requirement for celery.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-amqp/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    """Low-level AMQP client for Python (fork of amqplib)."""
+
+    homepage = "https://pypi.org/project/amqp/"
+    url      = "https://pypi.io/packages/source/a/amqp/amqp-2.4.1.tar.gz"
+
+    version('2.5.2', sha256='77f1aef9410698d20eaeac5b73a87817365f457a507d82edf292e12cbb83b08d')
+    version('2.4.2', sha256='043beb485774ca69718a35602089e524f87168268f0d1ae115f28b88d27f92d7')
+    version('2.4.1', sha256='6816eed27521293ee03aa9ace300a07215b11fee4e845588a9b863a7ba30addb')
+
+    depends_on('py-setuptools', type='build')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-amqp/package.py]: According to the repository this package [requires vine](https://github.com/celery/py-amqp/blob/master/requirements/default.txt)

Comment (koning) [reply var/spack/repos/builtin/packages/py-amqp/package.py]: Thanks, py-vine added.

File: var/spack/repos/builtin/packages/py-amqp/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+class PyAmqp(PythonPackage):
+    """Low-level AMQP client for Python (fork of amqplib)."""
+
+    homepage = "https://pypi.org/project/amqp/"
+    url      = "https://pypi.io/packages/source/a/amqp/amqp-2.4.1.tar.gz"
+
+    version('2.5.2', sha256='77f1aef9410698d20eaeac5b73a87817365f457a507d82edf292e12cbb83b08d')
+    version('2.4.2', sha256='043beb485774ca69718a35602089e524f87168268f0d1ae115f28b88d27f92d7')
+    version('2.4.1', sha256='6816eed27521293ee03aa9ace300a07215b11fee4e845588a9b863a7ba30addb')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-amqp/package.py]:
```python
depends_on('python@2.7:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-amqp/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on('py-setuptools', type='build')
+    depends_on('py-vine', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-amqp/package.py]:
```suggestion
    depends_on('py-vine@1.1.3:4.999', type=('build', 'run'))
```

**Comment by alalazo** [comment]
Argh apologies. Was about to merge and clicked the wrong button!

**Pull request merged**